sdk/storage/inc/common/shared_request_options.hpp

```diff
@@ -14,7 +15,7 @@ namespace Azure { namespace Storage {
      *        when the request is received by the service. If the timeout value
      *        elapses before the operation completes, the operation fails.
      */
-    int32_t Timeout = 0; // TODO: This is going to be changed into Nullable.
+    Azure::Core::Nullable<int32_t> Timeout;
```

Comment:
Customers should be controlling timeout via the context (which will work for ALL client libraries); not via an explicit field/parameter.

Reply:
This is a server-end timeout, which tells when the service should throw a timeout error. This is left as an optional parameter for other SDKs, is it a good approach to remove them?
However, I do agree that we can make the name to be a more obvious hint for a server-end timeout.

Reply:
In the Go SDK I had thee retry policy add this query parameter automatically based on the next cancellation time remaining.

Reply:
This indeed is more ideal. Is there a default value for this timeout in GO?